src/lib.rs

```diff
@@ -35,6 +35,22 @@ pub enum Error {
     PedersenParamsLen,
     #[error("Pedersen verification failed")]
     PedersenVerificationFail,
+    #[error("IVC verification failed")]
+    IVCVerificationFail,
+    #[error("R1CS instance is expected to not be relaxed")]
+    R1CSUnrelaxedFail,
+    #[error("Could not find the inner ConstraintSystem")]
+    NoInnerConstraintSystem,
+}
+
+#[macro_export]
+macro_rules! unwrap_or_return_err {
+    ( $e:expr, $err:expr ) => {
+        match $e {
+            Some(x) => x,
+            None => return $err,
+        }
+    };
```

Comment:
Isn't this the same as doing: `.map_err(|_| Err)?;` ??
I think we can avoid this macro.

Reply:
This was for `Option` so the `map_err` was not working, but I just realized that I could use `ok_or` and get rid of the `unwrap_or_return_err`. Updated to use `ok_or` and remove `unwrap_or_return_err`.

Reply:
Ohh true true!! My bad!

Reply:
nono no worries, your comment was very useful because made me realize that I didn't need the custom macro as `ok_or` exists ^^